### PR TITLE
read old node hex secrets

### PR DESCRIPTION
**Pull request by Emily**
@kesne

### CODE REVIEW

**Comment by kesne** [comment]
I WANT `--decrypt`!

**Comment by Emily** [comment]
@kesne I decided to do it as a separate PR, because it requires a much larger change than I originally thought

**Comment by Emily** [comment]
Well that's not entirely true, but my original change caused it to re-encrypt everything which seemed less than ideal.

File: src/encryption.rs [hunk 1]
```diff
@@ -108,11 +109,22 @@ impl Encryption<'_> {
...
-        if seckey_decoded.len() != SECRETKEYBYTES {
-            return Err("The secret key did not match the expected format.");
+        let seckey_decoded: Vec<u8>;
+
+        if sec_key.as_bytes().len() == 64 {
+            seckey_decoded = HEXUPPER_PERMISSIVE
+                .decode(sec_key.as_bytes())
+                .unwrap_or_default();
+
+            if seckey_decoded.len() != SECRETKEYBYTES {
```
Comment (kesne) [inline src/encryption.rs]: Can't this be outside of the if-else given it's the same for both?

Comment (Emily) [reply src/encryption.rs]: yes

**Review by kesne — APPROVED** [review]

**Pull request merged**